src/modules/router/services/router.transactions.service.ts

```diff
@@ -77,25 +82,29 @@ export class RouterTransactionService {
             throw new Error('Pair does not exist');
         }
 
-        const contract = await this.mxProxy.getRouterSmartContract();
         const endpointArgs: TypedValue[] = [
             BytesValue.fromUTF8(firstTokenID),
             BytesValue.fromUTF8(secondTokenID),
+            new AddressValue(Address.newFromBech32(sender)),
```

Comment:
this value is not actual accurate. Better would be to get it from the existing pair by the pair abi service and use that value here

Reply:
done